### ISSUE TITLE
HA Autodiscovery stopped working

### ISSUE THREAD

**Issue by VNDCK** [ridiekel/jeletask#28]
Hi Guy's! First of all a BIG thank you to Ridiekel & SanderM2 for making this happen. I Have this fully integrated in my HA setup and everything is (was :p) working just fine! Never touched the native old/ugly/slow Teletask App again, just GREAT. 

However, since a few days my setup stopped working and I'm not sure why. I've updated to the lastest version (of HA, Jeletask, Mosquitto) but this doesn't help. After restarting HA, it detects a new MQTT devices (my MQTT Broker), but after adding it, it doesn't detect any devices. My knowledge is to limited to see if this is an Mosquitto, HA or Jeletask issue? 

- I started the Docker container with the information below.

`sudo docker run --name jeletaskMQTT \
  -v "/setup/teletask/config.json:/teletask2mqtt/config.json" \
  -e TELETASK_HOST="192.168.1.100" \
  -e TELETASK_PORT="55957" \
  -e TELETASK_ID="TELETASK" \
  -e TELETASK_MQTT_HOST="192.168.1.99" \
  -e TELETASK_MQTT_USERNAME="mqtt" \
  -e TELETASK_MQTT_PASSWORD="xxxx" \
  --restart=always \
  ridiekel/jeletask2mqtt:latest-native`

- I've added the MQTT broker to my "configuration.yaml" file (not sure if this is really necessary because it auto-detects).

`mqtt:
  broker: 192.168.1.99
  username: mqtt
  password: xxx
  port: 1883`

- For testing, i'm using a very limited config file.

`{
  "type": "MICROS_PLUS",
  "componentsTypes": {
    "RELAY": [
      {
        "number": 1,
        "description": "Inkomhal, Spots"
      }
    ]
  }
}`

- With "MQTT Explorer" I can see all the MQTT messages.
- I've added some logs, because they show up once in a while (nothing changed on the Teletask side btw)

[logs.txt](https://github.com/ridiekel/jeletask/files/9899188/logs.txt)

**Comment by VNDCK** [comment]
Hi Guy's! Some more information. 

- Completely recreated all my docker containers (Home Assistant, Mosquitto Broker & Jeletask) > no change.
- Recreated the Jeletask container with version 3.3.6-native > Works! No error messages in the log files.
- Recreated with version "4.0.1-native" > Still works, with a very limited config.json. Afther changing my config.json (while Jeletask is running) I restarted HA & Jeletask > broke again + errors in log file.
- I do the same thing with 3.3.6 > Keeps working.

So, i'm now up-and-running on version 3.3.6. Is my workflow wrong?

Thx!

**Comment by ridiekel** [comment]
Glad you got it working. 

Can I close this isse?

**Comment by VNDCK** [comment]
> Glad you got it working.
> 
> Can I close this isse?

Yes, you can close the issue - I got it fixed. But there does seem to be some extra problems after 3.6 (going to 4.x).
